lib/jobs/ipmi-catalog.js

```diff
@@ -59,10 +63,20 @@ function ipmiCatalogJobFactory(
         var cmdResp = [];
 
         return waterline.obms.findByNode(self.nodeId, 'ipmi-obm-service', true)
-            .then(function (obmSettings) {
-                assert.object(obmSettings, 'No OBM service available.');
-                assert.ok(!_.isEmpty(obmSettings.config), 'No OBM service config available.');
+            .then(function(settings){
+                assert.object(settings, 'No OBM service available.');
+                assert.ok(!_.isEmpty(settings.config), 'No OBM service config available.');
+
+                if (settings.config.host && Constants.Regex.MacAddress.test(settings.config.host)) {
```

Comment:
We have already created a util function to handle this conversion, see https://github.com/RackHD/on-tasks/blob/master/lib/utils/job-utils/job-helper.js#L31, please change it to that function to avoid duplicated code.